commerce-test/src/testIntegration/java/com/liferay/commerce/shipment/test/CommerceShipmentTest.java

```diff
@@ -0,0 +1,1021 @@
+
+		BigDecimal value = BigDecimal.valueOf(RandomTestUtil.nextDouble());
+
+		CommerceOrder commerceOrder =
+			CommerceTestUtil.createCommerceOrderForShipping(
+				_company.getGroupId(), _user.getUserId(),
+				_commerceCurrency.getCommerceCurrencyId(),
+				commerceChannel.getSiteGroupId(), value);
+
+		_commerceOrders.add(commerceOrder);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			commerceOrder.getCommerceCurrency(), commerceChannel, null, null,
+			null, commerceOrder);
+
+		CommerceInventoryWarehouse commerceInventoryWarehouse =
+			_createCommerceInventoryWarehouse(
+				commerceChannel.getCommerceChannelId(), true);
+
+		int quantity = 10;
+
+		CommerceInventoryTestUtil.addCommerceInventoryWarehouseItem(
+			_user.getUserId(), commerceInventoryWarehouse, cpInstance.getSku(),
+			quantity);
+
+		int orderedQuantity = 1;
+
+		CommerceTestUtil.addCommerceOrderItem(
+			commerceOrder.getCommerceOrderId(), cpInstance.getCPInstanceId(),
+			orderedQuantity, commerceContext);
+
+		CommerceShipmentTestUtil.createOrderShipment(
+			commerceOrder.getGroupId(), commerceOrder.getCommerceOrderId(),
+			commerceInventoryWarehouse.getCommerceInventoryWarehouseId());
+
+		Assert.assertEquals(
+			false, _commerceShippingHelper.isShippable(commerceOrder));
+	}
+
+	@Test(expected = CommerceOrderStatusException.class)
+	public void testCreateShippingForCancelledOrder() throws Exception {
+		frutillaRule.scenario(
+			"Attach a shipment to an order with order status CANCELLED "
+		).given(
+			"An order is created"
+		).and(
+			"The order status is updated to CANCELLED"
+		).when(
+			"I attach the shipment to the items (one each)"
+		).then(
+			"An exception shall be raised"
+		);
+
+		CommerceChannel commerceChannel = _createCommerceChannel();
+
+		BigDecimal value = BigDecimal.valueOf(RandomTestUtil.nextDouble());
+
+		CommerceOrder commerceOrder =
+			CommerceTestUtil.createCommerceOrderForShipping(
+				_company.getGroupId(), _user.getUserId(),
+				_commerceCurrency.getCommerceCurrencyId(),
+				commerceChannel.getSiteGroupId(), value);
+
+		_commerceOrders.add(commerceOrder);
+
+		commerceOrder.setOrderStatus(
+			CommerceOrderConstants.ORDER_STATUS_CANCELLED);
+
+		_commerceOrderLocalService.updateCommerceOrder(commerceOrder);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			commerceOrder.getCommerceCurrency(), commerceChannel, null, null,
+			null, commerceOrder);
+
+		CommerceShipmentTestUtil.createEmptyOrderShipment(
+			commerceOrder.getGroupId(), commerceOrder.getCommerceOrderId());
+
+		_commerceOrderLocalService.checkoutCommerceOrder(
+			commerceOrder.getCommerceOrderId(), commerceContext,
+			ServiceContextTestUtil.getServiceContext(
+				commerceOrder.getGroupId()));
+	}
+
+	@Test
+	public void testCreateShippingForItemsOnly() throws Exception {
+		frutillaRule.scenario(
+			"Attach a shipment to the order items without a shipment to the " +
```

Comment:
`shipment to the order` should be
`shipment on the order`, or
`shipment against the order`